src/main/java/org/opensearch/ad/indices/ADIndexManagement.java

```diff
@@ -197,10 +195,7 @@ public void initDefaultResultIndexDirectly(ActionListener<CreateIndexResponse> a
     @Override
     public void initStateIndex(ActionListener<CreateIndexResponse> actionListener) {
         try {
-            // AD indices need RAW (e.g., we want users to be able to consume AD results as soon as possible and send out an alert if
-            // anomalies found).
-            Settings replicationSettings = Settings.builder().put(SETTING_REPLICATION_TYPE, DOCUMENT.name()).build();
-            CreateIndexRequest request = new CreateIndexRequest(ADCommonName.DETECTION_STATE_INDEX, replicationSettings)
+            CreateIndexRequest request = new CreateIndexRequest(ADCommonName.DETECTION_STATE_INDEX)
```

Comment:
@jackiehanyang : Thanks for working on this change.

With https://github.com/opensearch-project/OpenSearch/issues/8536, core now supports realtime reads with [get](https://opensearch.org/docs/latest/api-reference/document-apis/get-documents/)/[mget](https://opensearch.org/docs/latest/api-reference/document-apis/multi-get/) APIs for `SEGMENT` replication. Thus, please ensure search request needing RAW, are using get/mget APIs.

Reply:
@dreamer-89 What do you mean by using get/mget APIs for search request? We do have some search requests needing RAW.

Also, for realtime get/mget, do we need any changes in AD to use the new code in core?

Reply:
@kaituo : I meant if your plugin relies on get/mget APIs for data retrieval, then these APIs should automatically return latest data because this is handled on core. We recently added this support for SEGMENT replication enabled indices via https://github.com/opensearch-project/OpenSearch/issues/8536. Thus, as long as you are only using these APIs for RAW, no change is needed. 

>Also, for realtime get/mget, do we need any changes in AD to use the new code in core?

No

Also, please note DOCUMENT replication guarantees RAW on the write path by using `IMMEDIATE` or `WAIT_UNTIL` refresh policy, which ensure all replica shards are updated with same indexing operation. With SEGMENT replication this guarantee does not holds true. Thus, if you are using IMMEDIATE, WAIT_UNTIL refresh policy, then for follow up search requests (where you need RAW), you need to use either get/mget APIs or use _primary preference parameter. The former is preferred as later puts extra load on primary shard.

CC @mch2

Reply:
@dreamer-89 How do I replace search request with get/mget APIs?